### PR TITLE
[STORM-1766] - A better algorithm server rack selection for RAS

### PR DESCRIPTION
**Pull request by jerrypeng**

### CODE REVIEW

**Comment by knusbaum** [comment]
+1

File: storm-core/src/jvm/org/apache/storm/scheduler/resource/strategies/scheduling/DefaultResourceAwareStrategy.java [hunk 1]
```diff
@@ -45,6 +47,7 @@
 import org.apache.storm.scheduler.WorkerSlot;
 import org.apache.storm.scheduler.resource.Component;
 
+
```
Comment (redsanket) [inline storm-core/src/jvm/org/apache/storm/scheduler/resource/strategies/scheduling/DefaultResourceAwareStrategy.java]: extra line?

Comment (jerrypeng) [inline storm-core/src/jvm/org/apache/storm/scheduler/resource/strategies/scheduling/DefaultResourceAwareStrategy.java]: will remove

**Comment by jerrypeng** [comment]
@redsanket thanks for the review.  Do you have any other comments?

**Comment by ptgoetz** [comment]
+1

@jerrypeng Can you file a lira for updating the documentation if necessary?

**Comment by jerrypeng** [comment]
@ptgoetz I have created a jira:
https://issues.apache.org/jira/browse/STORM-1866

**Comment by ptgoetz** [comment]
@jerrypeng Did you merge this to any other branches, or just master?

**Comment by jerrypeng** [comment]
@ptgoetz just merged it into master why?

**Comment by ptgoetz** [comment]
@jerrypeng For tracking what goes into each branch/release. Github only gives us merge notifications for the branch a pull request targeted. If you had merged this to other branches, we wouldn't know unless we looked for it in other branches. That's why most of the time we add a comment noting which branches a patch was applied to. It saves a little time for other committers.

**Comment by jerrypeng** [comment]
@ptgoetz oh i see, thanks for letting me know! I will remember next time to put a comment in the jira regarding which branches i merged the corresponding PR to.

**Comment by HeartSaVioR** [comment]
@jerrypeng Since master branch is target to 2.0.0 and we don't have timeframe so it may be better to add it to 1.1.0 if you think it's not experimental feature.

**Pull request merged**